megamek/src/megamek/client/ui/swing/unitDisplay/UnitDisplay.java

```diff
@@ -119,14 +122,19 @@ public class UnitDisplay extends JPanel {
      *                  This could be null, if there is no ClientGUI, such as with
      *                  MekWars.
      */
-    public UnitDisplay(@Nullable ClientGUI clientgui) {
+    public UnitDisplay(ClientGUI clientgui) {
         this(clientgui, null);
     }
 
-    public UnitDisplay(@Nullable ClientGUI clientgui,
+    public UnitDisplay(ClientGUI clientgui, @Nullable MegaMekController controller) {
+        this(clientgui.getClient(), controller);
+        this.clientgui = clientgui;
+    }
+
+    public UnitDisplay(Client client,
             @Nullable MegaMekController controller) {
         super(new GridBagLayout());
-        this.clientgui = clientgui;
+        this.client = client;
```

Comment:
As indicated in the comment, this was left `Nullable` to support other clients such as MekWars.

Please fix this by re-adding the tag and adjusting the code accordingly. I'll bring MekWars support up with the rest of the Dev team to determine if we still wish to support it to some degree or try to merge that general idea into MegaMek.